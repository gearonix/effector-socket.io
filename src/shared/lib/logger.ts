const highlight = (color?: string) =>
  `background-color: ${color}; color: ${
    color ? '#000' : '#fff'
  }; padding-left: 4px; padding-right: 4px; font-weight: normal;`

export const createLogger = (isEnabled?: boolean) => {
  return (msg: string, method = 'method') => {
    if (isEnabled) {
      console.log(
        '%ceffector-socket.io' + `%c${msg}` + `%c[${method}]`,
        highlight('#4fb6ed'),
        highlight(),
        'color: #4fedde;'
      )
    }
  }
}
