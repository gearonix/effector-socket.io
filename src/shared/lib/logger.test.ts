/* eslint-disable max-len */

import { createLogger } from './logger'
import { highlight }    from './logger'

describe('highlight function', () => {
  let mockConsoleLog: jest.SpyInstance

  beforeEach(() => {
    mockConsoleLog = jest.spyOn(console, 'log').mockImplementation(() => {})
  })

  test('returns styles with provided color', () => {
    const result = highlight('#ff0000')
    expect(result).toBe(
      'background-color: #ff0000; color: #000; padding-left: 4px; padding-right: 4px; font-weight: normal;'
    )
  })

  test('returns default styles without color', () => {
    const result = highlight()

    expect(result).toBe(
      'background-color: none; color: #000; padding-left: 4px; padding-right: 4px; font-weight: normal;'
    )
  })

  test('logs messages when isEnabled is true', () => {
    const logger = createLogger(true)

    logger('hello', 'event')

    expect(mockConsoleLog).toHaveBeenCalledWith(
      '%ceffector-socket.io%chello%c[event]',
      'background-color: #4fb6ed; color: #000; padding-left: 4px; padding-right: 4px; font-weight: normal;',
      'background-color: none; color: #000; padding-left: 4px; padding-right: 4px; font-weight: normal;',
      'color: #4fedde;'
    )

    mockConsoleLog.mockRestore()
  })

  test('does not log messages when isEnabled is false', () => {
    const logger = createLogger(false)
    logger('Hello', 'Event', 'log')

    expect(mockConsoleLog).not.toHaveBeenCalled()

    mockConsoleLog.mockRestore()
  })
})
