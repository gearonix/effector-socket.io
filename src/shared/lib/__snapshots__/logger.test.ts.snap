// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight function logs messages when isEnabled is true 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "%ceffector-socket.io%chello%c[event]",
      "background-color: #4fb6ed; color: #000; padding-left: 4px; padding-right: 4px; font-weight: normal;",
      "background-color: none; color: #000; padding-left: 4px; padding-right: 4px; font-weight: normal;",
      "color: #4fedde;",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
